internal/k8s/controllers/config_controller.go

```diff
@@ -254,3 +271,44 @@ func (r *ConfigReconciler) getSecrets(ctx context.Context) (map[string]corev1.Se
 	}
 	return secretsMap, nil
 }
+
+func (r *ConfigReconciler) reportCondition(ctx context.Context, resultErr error, errType string) error {
+	if r.ConfigStateRef == nil {
+		return nil
+	}
+
+	owner := "configReconciler"
+	condition := metav1.Condition{
+		Type:    "configReconcilerValid",
+		Status:  metav1.ConditionTrue,
+		Reason:  errType,
+		Message: "",
+		// Always set to now (tracks last reconciliation, not last status change).
+		LastTransitionTime: metav1.Now(),
+	}
+
+	if resultErr != nil {
+		condition.Status = metav1.ConditionFalse
+		condition.Message = resultErr.Error()
+	}
+
+	configStatus := &metallbv1beta1.ConfigurationState{
```

Comment:
wouldn't it be better to have name / namespace as configreconciler fields here, and get before patching?
The reference feels a bit convoluted